### DISCUSSION TITLE
STARTTLS enabled, but throws error when initiating

### DISCUSSION THREAD

**Discussion by ajtonyb** [postalserver/postal#2581] [Help with using Postal] [answered]
I'm trying to set up starttls i have already turned it on inside postal.yml and specified the location. I put the .key and the .crt in the /config folder, which is specified inside the docker-compose. Everything seems to check out, but when I try to test it, I get the following error(s) in postal smtp logs:

![image](https://github.com/postalserver/postal/assets/115412163/719d43b6-9739-47a6-a54d-c48a1df1badd)

We use caddy and postal should be able to access the certs, since they have the same permissions as the postal.yml file and in the same /config folder.
I think my problem might be similar to #1512 , but not quite the same.

**Comment by ajtonyb** [comment]
Also, I get an error: **error:0A000126:SSL routines:ssl3_read_n:unexpected eof while reading**
to _openssl s_client -connect mysmtp.com:25 -starttls smtp_

It looks like an nginx problem.

**Comment by willpower232** [comment, marked as answer]
It sounds like the certificate file is corrupted or has been written weirdly by however it was generated.

You can try `postal bash smtp` and then `cat /config/cert.pem` or whatever file you have specified to check the contents look like a normal text file.

**Comment by ajtonyb** [comment]
The contents looked good, however I tried with a verified good cert-key pair and it didnt disconnect me.

**Reply by willpower232** [reply]
Perhaps a line ending or whitespace issue, things that aren't visible at a quick glance. Glad it got resolved.
